### PR TITLE
improve optimisation and structure

### PR DESCRIPTION
**Pull request by Luiggi33**
this pr aims to streamline the optimisations, with caching of all entity meta table funcs and also to improve uppon some minor style inconsistencies

### CODE REVIEW

File: lua/autorun/client/cfc_attention_monitoring_cl_init.lua [hunk 1]
```diff
@@ -61,33 +68,33 @@ local function formatAfkTime( rawTime )
 end
 
 local function drawIcon( ply )
-    if not IsValid( ply ) then
+    if not ply or ply == NULL then
```
Comment (sarahsturgeon) [inline lua/autorun/client/cfc_attention_monitoring_cl_init.lua]: Can you localize `NULL`? I forget if that works.

File: lua/autorun/client/cfc_attention_monitoring_cl_init.lua [hunk 2]
```diff
@@ -61,33 +68,33 @@ local function formatAfkTime( rawTime )
...
         table.RemoveByValue( trackedPlayers, ply )
         return
     end
+    if ply == localPlayer then return end
```
Comment (sarahsturgeon) [inline lua/autorun/client/cfc_attention_monitoring_cl_init.lua:75]: 👍

File: lua/autorun/client/cfc_attention_monitoring_cl_init.lua [hunk 3]
```diff
@@ -1,4 +1,6 @@
 local HasFocus = system.HasFocus
+local eyePos = EyePos
+local eyeAngles = EyeAngles
```
Comment (sarahsturgeon) [inline lua/autorun/client/cfc_attention_monitoring_cl_init.lua]: nit: You can localize these with the same casing - that's standard practice in our org anyway.

**Review by sarahsturgeon — COMMENTED** [review]
Thanks for the contribution!

LGTM - I'll let @wrefgtzweve make the final approval, as he's worked in here more recently than me.

I left a couple of nitpick comments, nothing major 👍 

Good work!

File: lua/autorun/client/cfc_attention_monitoring_cl_init.lua (hunk 3 above)
Comment (Luiggi33) [reply lua/autorun/client/cfc_attention_monitoring_cl_init.lua]: ah okay, not that familiar yet

File: lua/autorun/client/cfc_attention_monitoring_cl_init.lua (hunk 1 above)
Comment (Luiggi33) [reply lua/autorun/client/cfc_attention_monitoring_cl_init.lua]: i am actually not sure aswell xP

**Comment by Luiggi33** [comment]
Is there something that is holding back the merge of this?

File: lua/autorun/client/cfc_attention_monitoring_cl_init.lua [hunk 4]
```diff
@@ -61,24 +69,24 @@ local function formatAfkTime( rawTime )
 end
 
 local function drawIcon( ply )
-    if not IsValid( ply ) then
+    if not ply or ply == NULL then
```
Comment (wrefgtzweve) [inline lua/autorun/client/cfc_attention_monitoring_cl_init.lua]: not all NULL's are equal, this should an IsValid, can be :IsValid() likely

File: lua/autorun/client/cfc_attention_monitoring_cl_init.lua [hunk 5]
```diff
@@ -47,7 +55,7 @@ local function formatAfkTime( rawTime )
     local time = math_floor( rawTime )
     local hours = math_floor( ( time % 86400 ) / 3600 )
     local minutes = math_floor( ( time % 3600 ) / 60 )
-    --local seconds = math_floor( time % 60 )
+    -- local seconds = math_floor( time % 60 )
```
Comment (wrefgtzweve) [inline lua/autorun/client/cfc_attention_monitoring_cl_init.lua]: Can be removed

File: lua/autorun/client/cfc_attention_monitoring_cl_init.lua (hunk 4 above)
Comment (Luiggi33) [reply lua/autorun/client/cfc_attention_monitoring_cl_init.lua]: Could you be a bit more specific? From all my knowledge and tests, it does not matter in this context if I check IsValid or if it is NULL/nil or not

Comment (Luiggi33) [reply lua/autorun/client/cfc_attention_monitoring_cl_init.lua]: So I have done some searching and asking people, the == NULL check works for players. I know that vehicles and weapons have different NULLs, but for players NULL is correct. As long as we never wanna draw a icon above a vehicle for example, this should be more then fine

Comment (wrefgtzweve) [reply lua/autorun/client/cfc_attention_monitoring_cl_init.lua]: It's undefined behavior that shouldn't be relied upon, a localized meta function of IsValid() should be fine here

File: lua/autorun/client/cfc_attention_monitoring_cl_init.lua [hunk 6]
```diff
@@ -106,19 +113,20 @@ local function drawIcon( ply )
...
+
     for _, ply in ipairs( trackedPlayers ) do
         drawIcon( ply )
     end
 end
 
 hook.Add( "PostDrawTranslucentRenderables", "CFC_AttentionMonitor_AfkRenderElements", drawIcons )
 
-hook.Add( "EntityNetworkedVarChanged", "CFC_AttentionMonitor", function( ent, name, _, newval )
+hook.Add( "EntityNetworkedVarChanged", "CFC_AttentionMonitor", function( ent, name, _, newVal )
```
Comment (wrefgtzweve) [inline lua/autorun/client/cfc_attention_monitoring_cl_init.lua:129]: This will break, in the code below `newval` is used

Comment (Luiggi33) [reply lua/autorun/client/cfc_attention_monitoring_cl_init.lua:129]: This got lost in the merge conflict resolving, will fix in a minute

**Comment by wrefgtzweve** [comment]
Looks good to me once you fix the styling issue from the linter

**Review by wrefgtzweve — APPROVED** [review]
